semantics/executable-spec/src/Control/Iterate/SetAlgebraInternal.hs

```diff
@@ -1032,27 +1085,28 @@ compute (Dom (RExclude (Base MapR xs) (SetSingleton v))) = Sett(Map.foldlWithKey
 compute (Dom (RExclude (Base MapR xs) (Base SetR (Sett set)))) = Sett(Map.foldlWithKey' accum Set.empty xs)
    where accum ans k u = if not(Set.member u set) then Set.insert k ans else ans
 compute (Dom (DRestrict (SetSingleton v) (Base MapR xs))) = Sett(intersectMapSetFold accum xs (Set.singleton v) Set.empty)
-   where accum k u ans = if k==v then Set.insert k ans else ans
+   where accum k u ans = Set.insert k ans
 compute (Dom (DRestrict (Base SetR (Sett set)) (Base MapR xs))) = Sett(intersectMapSetFold accum xs set Set.empty)
-   where accum k u ans = if Set.member k set then Set.insert k ans else ans
-compute (Dom (DExclude (SetSingleton v) (Base MapR xs))) = Sett(intersectMapSetFold accum xs (Set.singleton v) Set.empty)
-   where accum k u ans = if not(k==v) then Set.insert k ans else ans
-compute (Dom (DExclude (Base SetR (Sett set)) (Base MapR xs))) = Sett(intersectMapSetFold accum xs set Set.empty)
-   where accum k u ans = if not(Set.member k set) then Set.insert k ans else ans
-compute (e@(Dom _)) = run(compile e)
+   where accum k u ans = Set.insert k ans
+
+compute (Dom (DExclude (SetSingleton v) (Base MapR xs))) = Sett(disjointMapSetFold accum xs (Set.singleton v) Set.empty)
+   where accum k u ans = Set.insert k ans
+compute (Dom (DExclude (Base SetR (Sett set)) (Base MapR xs))) = Sett(disjointMapSetFold accum xs set Set.empty)
+   where accum k u ans = Set.insert k ans
+compute (e@(Dom _)) = runSetExp e
 
 compute (Rng (Base SetR rel)) = Sett (Set.singleton ())
 compute (Rng (Singleton k v)) = Sett (Set.singleton v)
 compute (Rng (SetSingleton k)) = Sett (Set.singleton ())
 compute (Rng (Base rep rel)) = Sett(range rel)
-compute (e@(Rng _ )) = run(compile e)
+compute (e@(Rng _ )) = runSetExp e
 
 compute (DRestrict (Base SetR (Sett set)) (Base MapR m)) = Map.restrictKeys m set
 compute (DRestrict (SetSingleton k) (Base MapR m)) = Map.restrictKeys m (Set.singleton k)
 compute (DRestrict (Singleton k v) (Base MapR m)) = Map.restrictKeys m (Set.singleton k)
 compute (DRestrict (Dom (Base MapR x)) (Base MapR y)) = Map.intersection y x
 
-   -- This case inspired by set expression in EpochBoundary.hs
```

Comment:
I'm guessing you didn't mean this :)

Reply:
Actually, I did. There are so many tests, it is not clear why we need to identify where each pattern is inspired from.

Reply:
I don't follow, I just looks like the alignment is being broken, but it's not a big deal :)